web/pdf_viewer.js

```diff
@@ -1088,6 +1088,10 @@ class PDFViewer {
     this._currentScaleValue = newValue.toString();
 
     if (this.#isSameScale(newScale)) {
+      if (drawingDelay === -1 && this.#scaleTimeoutId) {
+        this.refresh();
+        return;
+      }
```

Comment:
I really don't understand what this block is actually for, and what it's trying to fix!?

It also seems to me that this could easily "break" the event-dispatching just below, and given all the different ways in which zooming is invoked I really can't tell if this is *generally* correct/safe unfortunately.